File: src/FN.Interfaces/Mapeadores/IMapeadorDeItensFinanceirosDeRecebimento.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Compartilhados.Interfaces.Core.Negocio;
using Compartilhados.Interfaces.FN.Negocio;

namespace FN.Interfaces.Mapeadores
{
    public interface IMapeadorDeItensFinanceirosDeRecebimento
    {
        void Insira(IItemLancamentoFinanceiroRecebimento Item);
        void Modifique(IItemLancamentoFinanceiroRecebimento Item);

        int ObtenhaQuantidadeDeItensFinanceiros(IFiltro filtro);

        IList<IItemLancamentoFinanceiroRecebimento> ObtenhaItensFinanceiros(IFiltro filtro, int quantidadeDeRegistros, int offSet);

    }
}

File: VersaoAjustesLocaweb/src/Compartilhados.Interfaces/FN/Mapeadores/IMapeadorDeItensFinanceirosDeRecebimento.vb
Imports Compartilhados.Interfaces.Core.Negocio
Imports Compartilhados.Interfaces.FN.Negocio

Namespace FN.Mapeadores

    Public Interface IMapeadorDeItensFinanceirosDeRecebimento

        Sub Insira(ByVal Item As IItemLancamentoFinanceiroRecebimento)
        Sub Modifique(ByVal Item As IItemLancamentoFinanceiroRecebimento)
        Function ObtenhaQuantidadeDeItensFinanceiros(ByVal filtro As IFiltro) As Integer
        Function ObtenhaItensFinanceiros(ByVal filtro As IFiltro, ByVal quantidadeDeRegistros As Integer, ByVal offSet As Integer) As IList(Of IItemLancamentoFinanceiroRecebimento)
        Function Obtenha(ByVal ID As Long) As IItemLancamentoFinanceiroRecebimento
        Sub Excluir(ByVal IdItemLancamentoFinanceiroRecebimento As Long)

    End Interface

End Namespace
